packages/nimble-components/src/dialog/styles.ts

```diff
@@ -18,11 +24,101 @@ import { themeBehavior } from '../utilities/style/theme';
 export const styles = css`
     ${display('grid')}
 
-    dialog {
+    dialog[open] {
+        display: flex;
+        flex-direction: column;
         background-color: ${applicationBackgroundColor};
-        border: ${borderWidth} solid ${popupBorderColor};
-        box-shadow: 0px 2px 3px ${popupBoxShadowColor};
-        max-width: 50%;
+        border: none;
+        box-shadow: 0px 4px 8px #0000004d;
+        padding: 0px;
+        width: 400px;
+        max-height: 600px;
+    }
+
+    header {
+        min-height: 48px;
+        padding: 24px 24px 0px 24px;
+        flex: none;
+        display: flex;
+        flex-direction: column;
+        gap: ${smallPadding};
+    }
+
+    :host([header-hidden]) header {
+        ${
+            /**
+             * Hide content visually while keeping it screen reader-accessible.
+             * Source: https://webaim.org/techniques/css/invisiblecontent/#techniques
+             * See discussion here: https://github.com/microsoft/fast/issues/5740#issuecomment-1068195035
+             */
+            ''
+        }
+        display: inline-block;
+        height: 1px;
+        width: 1px;
+        position: absolute;
+        margin: -1px;
+        clip: rect(1px, 1px, 1px, 1px);
+        clip-path: inset(50%);
+        overflow: hidden;
+        padding: 0;
+    }
+
+    slot[name='title'] {
+        font: ${titlePlus1Font};
+        color: ${titlePlus1FontColor};
+    }
+
+    slot[name='title']::slotted(*) {
+        overflow: hidden;
+        text-overflow: ellipsis;
+        white-space: nowrap;
+    }
```

Comment:
That makes sense. I can fix the flex layout issue by wrapping the slots in spans, but I can't get the overflow behavior I we without styling the slot contents. When I give the `span.title` element the ellipsis styling, it works when the title is given by the client as a `span` but not as a `div`.

Reply:
Playing around in devtools I think it can be tweaked a bit. Conceptually we should say that users give us a "text run" instead of a paragraph, ie a line in a paragraph. So they should give us something `display: inline`. We will put that into a container that is block level and controls the wrapping, sizing, etc.

So the template would change to:
```
            <header id="header">
                <div class="title">
                    <slot name="title"></slot>
                </div>
                <div class="subtitle">
                   <slot name="subtitle"></slot>
                </div>
            </header>

```

and users give:

```
<span slot="title">Dialog title</span>
<span slot="subtitle">Dialog subtitle</span>
```

And we can style the block level div.title
```
white-space: nowrap;
overflow: hidden;
text-overflow: ellipsis;
```

Reply:
This may be a pattern we want to check is being done consistently, That (where appropriate) we are treating slotted content as inline text instead of relying on the user to configure block level settings correctly (I imagine it's relevant for button content, radio, etc, but those may also rely on intrinsic sizing, ie fit-content). Not sure if there is a specific action to take, mostly thoughts.

Reply:
Just spot checking a few things (button content, radio, switch), I didn't see anything else that behaves differently based on whether a `<div>` or a `<span>` is slotter (or if the content was wrapped in a `<div>` or a `<span>` rather than text content).

To make sure I understand what you are suggesting -- we should style the component with the assumption that an `inline` component is slotted and then not care if the styling is off when a `block` component is slotted? Should we document that somewhere?